lib/origin_simulator.ex

```diff
@@ -2,47 +2,52 @@ defmodule OriginSimulator do
   use Plug.Router
   alias OriginSimulator.{Payload, Recipe, Simulation, PlugResponseCounter, Counter}
 
-  @default_route Recipe.default_route()
-
   plug(PlugResponseCounter)
   plug(:match)
   plug(:dispatch)
 
-  get "/status" do
+  @admin_domain Application.get_env(:origin_simulator, :admin_domain)
+
+  get "/:admin/status" when admin == @admin_domain do
```

Comment:
```suggestion
  get "/#{@admin_domain}/status" do
```
Would something like ^this be a more readable?

My suggestion might be bad Elixir style though, I'm looking at it as a rubyist 😄

Reply:
We could move these admin routes to a separate router using https://hexdocs.pm/plug/Plug.Router.html#forward/2, and have something like this:
```
forward "/#{@admin_domain}", to: AdminRouter
```
then in AdminRouter only have the `/status`, `/clear_current_count` etc...

Reply:
Good ideas @james-bowers, @woodyblah, shall do that.  I was trying to refer to a private function in the route previously and it throws `undefined function admin_domain/0` (for the macro), no such problem for module variable.

```ex
get "/#{admin_domain()}/status" do
```

Reply:
There's now a separate router for the admin routes, however, I hard-coded the forward route:
```ex
forward "/_admin", to: OriginSimulator.AdminRouter
```
This is because Plug's forward macro is inferior to the other macros such as `get`, `post`. It doesn't provide string interpolation (`"#{@admin_domain}"`) and guards for any atomic URL key. Check the macro and compare it with the ones above that are compiled. 

https://github.com/elixir-plug/plug/blob/v1.8.3/lib/plug/router.ex#L424